homeassistant/components/icloud/account.py

```diff
@@ -105,12 +106,22 @@ def setup(self) -> None:
             return
 
         user_info = None
+        api_devices = {}
         try:
+            api_devices = self.api.devices
             # Gets device owners infos
-            user_info = self.api.devices.response["userInfo"]
-        except PyiCloudNoDevicesException:
+            user_info = api_devices.response["userInfo"]
+        except (KeyError, PyiCloudNoDevicesException):
             _LOGGER.error("No iCloud device found")
-            return
+            raise ConfigEntryNotReady
+
+        if not api_devices:
```

Comment:
This can't happen because line 113 would have crashed already ?

Reply:
Indeed